include/options.php

```diff
@@ -0,0 +1,647 @@
+ * @phpstan-type ListDataKeys 'post_types'|'sync'|'taxonomies'
+ * @phpstan-type StringDataKeys 'default_lang'|'previous_version'|'version'
+ * @phpstan-type NavMenuType array<
+ *     non-falsy-string,
+ *     array<
+ *         non-falsy-string,
+ *         array<
+ *             non-falsy-string,
+ *             int<0, max>
+ *         >
+ *     >
+ * >
+ * @phpstan-type OptionsData array{
+ *     browser: bool,
+ *     default_lang: string,
+ *     domains: array<non-falsy-string, non-falsy-string>,
+ *     first_activation: int<0, max>,
+ *     force_lang: int-mask<0, 1, 2, 3>,
+ *     hide_default: bool,
+ *     media: array<non-falsy-string, 1>,
+ *     media_support: bool,
+ *     nav_menus: NavMenuType,
+ *     post_types: list<non-falsy-string>,
+ *     previous_version: string,
+ *     redirect_lang: bool,
+ *     rewrite: bool,
+ *     sync: list<non-falsy-string>,
+ *     taxonomies: list<non-falsy-string>,
+ *     uninstall: bool,
+ *     version: string
+ * }
+ * @phpstan-type DefaultOptionsData array{
+ *     browser: false,
+ *     default_lang: '',
+ *     domains: array<non-falsy-string, non-falsy-string>,
+ *     first_activation: 0,
+ *     force_lang: 0,
+ *     hide_default: false,
+ *     media: array<non-falsy-string, 1>,
+ *     media_support: false,
+ *     nav_menus: NavMenuType,
+ *     post_types: list<non-falsy-string>,
+ *     previous_version: '',
+ *     redirect_lang: false,
+ *     rewrite: false,
+ *     sync: list<non-falsy-string>,
+ *     taxonomies: list<non-falsy-string>,
+ *     uninstall: false,
+ *     version: ''
+ * }
+ * @phpstan-type ResetOptionsData array{
+ *     browser: false,
+ *     default_lang: '',
+ *     domains: array<non-falsy-string, non-falsy-string>,
+ *     first_activation: positive-int,
+ *     force_lang: 1,
+ *     hide_default: true,
+ *     media: array<non-falsy-string, 1>,
+ *     media_support: false,
+ *     nav_menus: NavMenuType,
+ *     post_types: list<non-falsy-string>,
+ *     previous_version: '',
+ *     redirect_lang: false,
+ *     rewrite: true,
+ *     sync: list<non-falsy-string>,
+ *     taxonomies: list<non-falsy-string>,
+ *     uninstall: false,
+ *     version: non-falsy-string
+ * }
+ */
+class PLL_Options implements ArrayAccess, Countable, Iterator, JsonSerializable {
+
+	const OPTION_NAME = 'polylang';
+
+	/**
+	 * @var mixed[]
+	 * @phpstan-var DefaultOptionsData
+	 */
+	const DEFAULTS = array(
+		'browser'          => false,
+		'default_lang'     => '',
+		'domains'          => array(),
+		'first_activation' => 0,
+		'force_lang'       => 0,
+		'hide_default'     => false,
+		'media'            => array(),
+		'media_support'    => false,
+		'nav_menus'        => array(),
+		'post_types'       => array(),
+		'previous_version' => '',
+		'redirect_lang'    => false,
+		'rewrite'          => false,
+		'sync'             => array(),
+		'taxonomies'       => array(),
+		'uninstall'        => false,
+		'version'          => '',
```

Comment:
With #1267 in mind, I believe that this should default to POLYLANG_VERSION.

Reply:
I didn't do that so we can detect the first install.
I'll double-check if it's still necessary.

Reply:
I believe we should keep it empty:
- https://github.com/polylang/polylang/pull/1237/files#diff-5214ac2ea5eb9c1915795e0fe082fab413112c144ce3009764b0d86179d48b69R179
- https://github.com/polylang/polylang/pull/1237/files#diff-7bcdbc1cb24cba2bfcef22987b4a3bc3910e4972ac2cbf5a219a960327d8c939R168
- https://github.com/polylang/polylang/pull/1237/files#diff-7b56c8ba8d31fce73086ef07085559611b4219b1c2d6e723a623f71b581f2745R107
- https://github.com/polylang/polylang/pull/1237/files#diff-a36b4a68ce7c6a8aa48b0b5f5ec135b11f757493710d614497b79a66302e7849R91